{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None \t; Convert MAF to Genomic Intervals\n",
      "None \t; Convert GFF to Feature Location Index\n",
      "None \t; Convert SAM to BAM\n",
      "None \t; Convert FASTA to len file\n",
      "None \t; Convert FASTA to Bowtie color space Index\n",
      "None \t; Convert FASTA to 2bit\n",
      "None \t; Convert Genomic Intervals To Strict BED12\n",
      "None \t; Convert BED to GFF\n",
      "None \t; Convert MAF to Fasta\n",
      "None \t; Convert Genomic Intervals To Strict BED\n",
      "None \t; Convert FASTA to Tabular\n",
      "None \t; Convert Genomic Intervals To Coverage\n",
      "None \t; Wiggle to Interval\n",
      "None \t; Import History\n",
      "None \t; Convert FASTA to Bowtie base space Index\n",
      "None \t; Convert Genomic Intervals To BED\n",
      "None \t; Convert BED to Feature Location Index\n",
      "None \t; Convert Genomic Intervals To Strict BED6\n",
      "None \t; Convert plink pbed to ld reduced format\n",
      "None \t; Convert Len file to Linecount\n",
      "None \t; Convert tabular to dbnsfp\n",
      "None \t; Convert GFF to BED\n",
      "None \t; Export History\n",
      "None \t; Convert lped to fped\n",
      "None \t; Convert plink pbed to linkage lped\n",
      "None \t; Set External Metadata\n",
      "None \t; Convert lped to plink pbed\n",
      "None \t; Convert Picard Interval List to BED6\n",
      "BEDTools \t; Convert from BAM to BED\n",
      "BEDTools \t; Create a BedGraph of genome coverage\n",
      "BEDTools \t; Count intervals in one file overlapping intervals in another file\n",
      "BEDTools \t; Intersect multiple sorted BED files\n",
      "BEDTools \t; Merge BedGraph files\n",
      "BEDTools \t; Create a histogram of genome coverage\n",
      "BEDTools \t; Intersect BAM alignments with intervals in another files\n",
      "Convert Formats \t; MAF to BED\n",
      "Convert Formats \t; MAF to Interval\n",
      "Convert Formats \t; MAF to FASTA\n",
      "Convert Formats \t; GFF-to-BED\n",
      "Convert Formats \t; BED-to-bigBed\n",
      "Convert Formats \t; Wig/BedGraph-to-bigWig\n",
      "Convert Formats \t; SFF converter\n",
      "Convert Formats \t; BED-to-GFF\n",
      "EMBOSS \t; infoseq\n",
      "EMBOSS \t; helixturnhelix\n",
      "EMBOSS \t; msbar\n",
      "EMBOSS \t; pepinfo\n",
      "EMBOSS \t; digest\n",
      "EMBOSS \t; pepwindowall\n",
      "EMBOSS \t; octanol\n",
      "EMBOSS \t; banana\n",
      "EMBOSS \t; geecee\n",
      "EMBOSS \t; oddcomp\n",
      "EMBOSS \t; pasteseq\n",
      "EMBOSS \t; sixpack\n",
      "EMBOSS \t; prettyseq\n",
      "EMBOSS \t; diffseq\n",
      "EMBOSS \t; chips\n",
      "EMBOSS \t; fuzzpro\n",
      "EMBOSS \t; vectorstrip\n",
      "EMBOSS \t; seqmatchall\n",
      "EMBOSS \t; wordcount\n",
      "EMBOSS \t; einverted\n",
      "EMBOSS \t; nthseq\n",
      "EMBOSS \t; checktrans\n",
      "EMBOSS \t; cpgreport\n",
      "EMBOSS \t; pepcoil\n",
      "EMBOSS \t; charge\n",
      "EMBOSS \t; coderet\n",
      "EMBOSS \t; degapseq\n",
      "EMBOSS \t; marscan\n",
      "EMBOSS \t; maskfeat\n",
      "EMBOSS \t; supermatcher\n",
      "EMBOSS \t; equicktandem\n",
      "EMBOSS \t; tranalign\n",
      "EMBOSS \t; cai custom\n",
      "EMBOSS \t; noreturn\n",
      "EMBOSS \t; preg\n",
      "EMBOSS \t; transeq\n",
      "EMBOSS \t; seqret\n",
      "EMBOSS \t; showfeat\n",
      "EMBOSS \t; extractfeat\n",
      "EMBOSS \t; cai\n",
      "EMBOSS \t; dreg\n",
      "EMBOSS \t; twofeat\n",
      "EMBOSS \t; btwisted\n",
      "EMBOSS \t; newseq\n",
      "EMBOSS \t; plotcon\n",
      "EMBOSS \t; garnier\n",
      "EMBOSS \t; trimest\n",
      "EMBOSS \t; dotpath\n",
      "EMBOSS \t; polydot\n",
      "EMBOSS \t; wobble\n",
      "EMBOSS \t; dan\n",
      "EMBOSS \t; sirna\n",
      "EMBOSS \t; est2genome\n",
      "EMBOSS \t; fuzznuc\n",
      "EMBOSS \t; freak\n",
      "EMBOSS \t; pepwindow\n",
      "EMBOSS \t; fuzztran\n",
      "EMBOSS \t; newcpgseek\n",
      "EMBOSS \t; codcmp\n",
      "EMBOSS \t; newcpgreport\n",
      "EMBOSS \t; sigcleave\n",
      "EMBOSS \t; union\n",
      "EMBOSS \t; getorf\n",
      "EMBOSS \t; pepstats\n",
      "EMBOSS \t; skipseq\n",
      "EMBOSS \t; splitter\n",
      "EMBOSS \t; tcode\n",
      "EMBOSS \t; maskseq\n",
      "EMBOSS \t; patmatdb\n",
      "EMBOSS \t; cpgplot\n",
      "EMBOSS \t; notseq\n",
      "EMBOSS \t; water\n",
      "EMBOSS \t; trimseq\n",
      "EMBOSS \t; syco\n",
      "EMBOSS \t; compseq\n",
      "EMBOSS \t; merger\n",
      "EMBOSS \t; dottup\n",
      "EMBOSS \t; pepnet\n",
      "EMBOSS \t; prettyplot\n",
      "EMBOSS \t; shuffleseq\n",
      "EMBOSS \t; cusp\n",
      "EMBOSS \t; matcher\n",
      "EMBOSS \t; hmoment\n",
      "EMBOSS \t; tmap\n",
      "EMBOSS \t; cutseq\n",
      "EMBOSS \t; epestfind\n",
      "EMBOSS \t; biosed\n",
      "EMBOSS \t; antigenic\n",
      "EMBOSS \t; needle\n",
      "EMBOSS \t; extractseq\n",
      "EMBOSS \t; revseq\n",
      "EMBOSS \t; iep\n",
      "EMBOSS \t; pepwheel\n",
      "EMBOSS \t; megamerger\n",
      "EMBOSS \t; lindna\n",
      "EMBOSS \t; palindrome\n",
      "EMBOSS \t; plotorf\n",
      "EMBOSS \t; etandem\n",
      "EMBOSS \t; chaos\n",
      "EMBOSS \t; textsearch\n",
      "EMBOSS \t; primersearch\n",
      "EMBOSS \t; descseq\n",
      "EMBOSS \t; cirdna\n",
      "EMBOSS \t; isochore\n",
      "EMBOSS \t; wordmatch\n",
      "EMBOSS \t; backtranseq\n",
      "EMBOSS \t; dotmatcher\n",
      "Evolution \t; Mutate Codons\n",
      "Extract Features \t; Gene BED To Exon/Intron/Codon BED\n",
      "FASTA manipulation \t; Filter sequences by length\n",
      "FASTA manipulation \t; Tabular-to-FASTA\n",
      "FASTA manipulation \t; FASTA Width\n",
      "FASTA manipulation \t; FASTA-to-Tabular\n",
      "FASTA manipulation \t; Compute sequence length\n",
      "FASTA manipulation \t; Collapse\n",
      "FASTA manipulation \t; RNA/DNA\n",
      "FASTA manipulation \t; Concatenate\n",
      "Fetch Alignments \t; Filter MAF\n",
      "Fetch Alignments \t; Stitch Gene blocks\n",
      "Fetch Alignments \t; Filter MAF blocks\n",
      "Fetch Alignments \t; Filter MAF blocks\n",
      "Fetch Alignments \t; Stitch MAF blocks\n",
      "Fetch Alignments \t; Extract MAF blocks\n",
      "Fetch Alignments \t; Reverse Complement\n",
      "Fetch Alignments \t; Extract Pairwise MAF blocks\n",
      "Fetch Alignments \t; Extract MAF by block number\n",
      "Fetch Alignments \t; Join MAF blocks\n",
      "Fetch Alignments \t; MAF Coverage Stats\n",
      "Fetch Sequences \t; Extract Genomic DNA\n",
      "Filter and Sort \t; Filter GFF data by attribute\n",
      "Filter and Sort \t; Select\n",
      "Filter and Sort \t; Histogram\n",
      "Filter and Sort \t; Sort\n",
      "Filter and Sort \t; Extract features\n",
      "Filter and Sort \t; Filter GFF data by feature count\n",
      "Filter and Sort \t; Filter GTF data by attribute values_list\n",
      "Filter and Sort \t; Filter\n",
      "Genome Diversity \t; Flanking Sequence\n",
      "Genome Diversity \t; Pick Primers\n",
      "Genome Diversity \t; Per-SNP FSTs\n",
      "Genome Diversity \t; Draw variants\n",
      "Genome Diversity \t; Rank Pathways\n",
      "Genome Diversity \t; Convert\n",
      "Genome Diversity \t; Differential Cleavage\n",
      "Genome Diversity \t; Inbreeding and kinship\n",
      "Genome Diversity \t; Sample SNPs\n",
      "Genome Diversity \t; Remarkable Intervals\n",
      "Genome Diversity \t; Pathway Image\n",
      "Genome Diversity \t; Filter SNPs\n",
      "Genome Diversity \t; Restore Attributes\n",
      "Genome Diversity \t; Get Pathways\n",
      "Genome Diversity \t; Specify Individuals\n",
      "Genome Diversity \t; Diversity\n",
      "Genome Diversity \t; gd_snp to VCF\n",
      "Genome Diversity \t; Reorder individuals\n",
      "Genome Diversity \t; Overall FST\n",
      "Genome Diversity \t; RAxML\n",
      "Genome Diversity \t; Coverage Distributions\n",
      "Genome Diversity \t; Nucleotide Diversity\n",
      "Genome Diversity \t; Matings\n",
      "Genome Diversity \t; Phylip\n",
      "Genome Diversity \t; Pairs sequenced\n",
      "Genome Diversity \t; Aggregate Individuals\n",
      "Genome Diversity \t; Close relatives\n",
      "Genome Diversity \t; PCA\n",
      "Genome Diversity \t; Rank Terms\n",
      "Genome Diversity \t; Ancestry\n",
      "Genome Diversity \t; Phylogenetic Tree\n",
      "Genome Diversity \t; Admixture\n",
      "Genome Diversity \t; Cluster KEGG\n",
      "Genome Diversity \t; Founders sequenced\n",
      "Genome Diversity \t; Prepare Input\n",
      "Genome Diversity \t; Make File\n",
      "Genome Diversity \t; Population Complexity\n",
      "Get Data \t; UCSC Main\n",
      "Get Data \t; GenomeSpace import\n",
      "Get Data \t; WormBase\n",
      "Get Data \t; modENCODE worm\n",
      "Get Data \t; Ratmine\n",
      "Get Data \t; UCSC Archaea\n",
      "Get Data \t; Upload File\n",
      "Get Data \t; modENCODE fly\n",
      "Get Data \t; Flymine\n",
      "Get Data \t; ZebrafishMine\n",
      "Get Data \t; MouseMine\n",
      "Get Data \t; EBI SRA\n",
      "Get Data \t; modENCODE modMine\n",
      "Get Data \t; GrameneMart\n",
      "Get Data \t; BioMart\n",
      "Get Data \t; EuPathDB\n",
      "Get Data \t; YeastMine\n",
      "Get Genomic Scores \t; Wiggle-to-Interval\n",
      "Get Genomic Scores \t; Aggregate datapoints\n",
      "Graph/Display Data \t; Plotting tool\n",
      "Graph/Display Data \t; Scatterplot\n",
      "Graph/Display Data \t; Build custom track\n",
      "Graph/Display Data \t; Histogram\n",
      "Graph/Display Data \t; VCF to MAF Custom Track\n",
      "Graph/Display Data \t; Boxplot\n",
      "Graph/Display Data \t; GMAJ\n",
      "Join, Subtract and Group \t; Group\n",
      "Join, Subtract and Group \t; Compare two Datasets\n",
      "Join, Subtract and Group \t; Join two Datasets\n",
      "Join, Subtract and Group \t; Subtract Whole Dataset\n",
      "Lift-Over \t; Convert genome coordinates\n",
      "Metagenomic analyses \t; Find diagnostic hits\n",
      "Metagenomic analyses \t; Poisson two-sample test\n",
      "Metagenomic analyses \t; Find lowest diagnostic rank\n",
      "Metagenomic analyses \t; Draw phylogeny\n",
      "Metagenomic analyses \t; Summarize taxonomy\n",
      "Metagenomic analyses \t; Fetch taxonomic representation\n",
      "Motif Tools \t; Sequence Logo\n",
      "Multiple Alignments \t; ClustalW\n",
      "NGS: BAM Tools \t; Split\n",
      "NGS: BAM Tools \t; Filter\n",
      "NGS: BAM Tools \t; Convert, Merge, Randomize\n",
      "NGS: GATK Tools (beta) \t; Combine Variants\n",
      "NGS: GATK Tools (beta) \t; Unified Genotyper\n",
      "NGS: GATK Tools (beta) \t; Apply Variant Recalibration\n",
      "NGS: GATK Tools (beta) \t; Variant Annotator\n",
      "NGS: GATK Tools (beta) \t; Print Reads\n",
      "NGS: GATK Tools (beta) \t; Realigner Target Creator\n",
      "NGS: GATK Tools (beta) \t; Analyze Covariates\n",
      "NGS: GATK Tools (beta) \t; Select Variants\n",
      "NGS: GATK Tools (beta) \t; Eval Variants\n",
      "NGS: GATK Tools (beta) \t; Indel Realigner\n",
      "NGS: GATK Tools (beta) \t; Count Covariates\n",
      "NGS: GATK Tools (beta) \t; Table Recalibration\n",
      "NGS: GATK Tools (beta) \t; Variant Filtration\n",
      "NGS: GATK Tools (beta) \t; Variant Recalibrator\n",
      "NGS: GATK Tools (beta) \t; Validate Variants\n",
      "NGS: GATK Tools (beta) \t; Depth of Coverage\n",
      "NGS: Mapping \t; BWA-MEM\n",
      "NGS: Mapping \t; Map with Bowtie for Illumina\n",
      "NGS: Mapping \t; Map with BWA for Illumina\n",
      "NGS: Mapping \t; Parse blast XML output\n",
      "NGS: Mapping \t; Bowtie2\n",
      "NGS: Mapping \t; Lastz\n",
      "NGS: Mapping \t; BWA\n",
      "NGS: Mapping \t; Megablast\n",
      "NGS: Peak Calling \t; SICER\n",
      "NGS: Peak Calling \t; MACS\n",
      "NGS: Picard \t; MarkDuplicates\n",
      "NGS: Picard \t; CollectInsertSizeMetrics\n",
      "NGS: Picard \t; CollectBaseDistributionByCycle\n",
      "NGS: Picard \t; SortSam\n",
      "NGS: Picard \t; FilterSamReads\n",
      "NGS: Picard \t; AddCommentsToBam\n",
      "NGS: Picard \t; SamToFastq\n",
      "NGS: Picard \t; EstimateLibraryComplexity\n",
      "NGS: Picard \t; CollectWgsMetrics\n",
      "NGS: Picard \t; CollectGcBiasMetrics\n",
      "NGS: Picard \t; QualityScoreDistribution\n",
      "NGS: Picard \t; ReorderSam\n",
      "NGS: Picard \t; MarkDuplicatesWithMateCigar\n",
      "NGS: Picard \t; CleanSam\n",
      "NGS: Picard \t; MergeSamFiles\n",
      "NGS: Picard \t; MergeBamAlignment\n",
      "NGS: Picard \t; RevertSam\n",
      "NGS: Picard \t; Downsample SAM/BAM\n",
      "NGS: Picard \t; FixMateInformation\n",
      "NGS: Picard \t; BedToIntervalList\n",
      "NGS: Picard \t; MeanQualityByCycle\n",
      "NGS: Picard \t; AddOrReplaceReadGroups\n",
      "NGS: Picard \t; Collect Alignment Summary Metrics\n",
      "NGS: Picard \t; ReplaceSamHeader\n",
      "NGS: Picard \t; RevertOriginalBaseQualitiesAndAddMateCigar\n",
      "NGS: Picard \t; CollectRnaSeqMetrics\n",
      "NGS: Picard \t; ValidateSamFile\n",
      "NGS: Picard \t; NormalizeFasta\n",
      "NGS: Picard \t; FastqToSam\n",
      "NGS: Picard (beta) \t; SAM to FASTQ\n",
      "NGS: Picard (beta) \t; SAM/BAM Alignment Summary Metrics\n",
      "NGS: Picard (beta) \t; Insertion size metrics\n",
      "NGS: Picard (beta) \t; Estimate Library Complexity\n",
      "NGS: Picard (beta) \t; Add or Replace Groups\n",
      "NGS: Picard (beta) \t; Replace SAM/BAM Header\n",
      "NGS: Picard (beta) \t; BAM Index Statistics\n",
      "NGS: Picard (beta) \t; FASTQ to BAM\n",
      "NGS: Picard (beta) \t; Mark Duplicate reads\n",
      "NGS: Picard (beta) \t; Paired Read Mate Fixer\n",
      "NGS: Picard (beta) \t; SAM/BAM Hybrid Selection Metrics\n",
      "NGS: Picard (beta) \t; SAM/BAM GC Bias Metrics\n",
      "NGS: Picard (beta) \t; Reorder SAM/BAM\n",
      "NGS: QC and manipulation \t; Trim sequences\n",
      "NGS: QC and manipulation \t; FASTQ splitter\n",
      "NGS: QC and manipulation \t; FASTQ to FASTA\n",
      "NGS: QC and manipulation \t; Filter by quality\n",
      "NGS: QC and manipulation \t; Rename sequences\n",
      "NGS: QC and manipulation \t; FASTQ to Tabular\n",
      "NGS: QC and manipulation \t; Build base quality distribution\n",
      "NGS: QC and manipulation \t; Draw quality score boxplot\n",
      "NGS: QC and manipulation \t; FASTQ Trimmer\n",
      "NGS: QC and manipulation \t; FASTQ joiner\n",
      "NGS: QC and manipulation \t; Draw quality score boxplot\n",
      "NGS: QC and manipulation \t; Remove sequencing artifacts\n",
      "NGS: QC and manipulation \t; Barcode Splitter\n",
      "NGS: QC and manipulation \t; FastQC\n",
      "NGS: QC and manipulation \t; FASTQ Groomer\n",
      "NGS: QC and manipulation \t; Quality format converter\n",
      "NGS: QC and manipulation \t; Clip\n",
      "NGS: QC and manipulation \t; Draw nucleotides distribution chart\n",
      "NGS: QC and manipulation \t; FASTQ Masker\n",
      "NGS: QC and manipulation \t; Compute quality statistics\n",
      "NGS: QC and manipulation \t; Manipulate FASTQ\n",
      "NGS: QC and manipulation \t; FASTQ to FASTA\n",
      "NGS: QC and manipulation \t; Reverse-Complement\n",
      "NGS: QC and manipulation \t; Compute quality statistics\n",
      "NGS: QC and manipulation \t; Combine FASTA and QUAL\n",
      "NGS: QC and manipulation \t; Filter FASTQ\n",
      "NGS: QC and manipulation \t; Tabular to FASTQ\n",
      "NGS: QC and manipulation \t; Select high quality segments\n",
      "NGS: QC and manipulation \t; FASTQ Quality Trimmer\n",
      "NGS: QC and manipulation \t; Convert\n",
      "NGS: QC and manipulation \t; FASTQ Summary Statistics\n",
      "NGS: RNA-seq \t; Cuffcompare\n",
      "NGS: RNA-seq \t; Cufflinks\n",
      "NGS: RNA-seq \t; Tophat\n",
      "NGS: RNA-seq \t; Tophat2\n",
      "NGS: RNA-seq \t; Tophat\n",
      "NGS: RNA-seq \t; Cuffdiff\n",
      "NGS: RNA-seq \t; Cuffdiff\n",
      "NGS: RNA-seq \t; Cuffmerge\n",
      "NGS: RNA-seq \t; Cuffmerge\n",
      "NGS: RNA-seq \t; Filter Combined Transcripts\n",
      "NGS: RNA-seq \t; Cuffquant\n",
      "NGS: RNA-seq \t; Cuffmerge\n",
      "NGS: RNA-seq \t; Tophat Fusion Post\n",
      "NGS: RNA-seq \t; gffread\n",
      "NGS: RNA-seq \t; Cuffdiff\n",
      "NGS: RNA-seq \t; Cuffdiff\n",
      "NGS: RNA-seq \t; StringTie\n",
      "NGS: RNA-seq \t; Cuffnorm\n",
      "NGS: SAMtools \t; Slice BAM\n",
      "NGS: SAMtools \t; Convert SAM\n",
      "NGS: SAMtools \t; Reheader\n",
      "NGS: SAMtools \t; BAM-to-SAM\n",
      "NGS: SAMtools \t; Pileup-to-Interval\n",
      "NGS: SAMtools \t; Filter SAM or BAM, output SAM or BAM\n",
      "NGS: SAMtools \t; BedCov\n",
      "NGS: SAMtools \t; MPileup\n",
      "NGS: SAMtools \t; flagstat\n",
      "NGS: SAMtools \t; SAM-to-BAM\n",
      "NGS: SAMtools \t; Flagstat\n",
      "NGS: SAMtools \t; Sort\n",
      "NGS: SAMtools \t; Slice\n",
      "NGS: SAMtools \t; CalMD\n",
      "NGS: SAMtools \t; RmDup\n",
      "NGS: SAMtools \t; bcftools view\n",
      "NGS: SAMtools \t; SAM-to-BAM\n",
      "NGS: SAMtools \t; SAM-to-BAM\n",
      "NGS: SAMtools \t; Split\n",
      "NGS: SAMtools \t; Filter pileup\n",
      "NGS: SAMtools \t; BAM-to-SAM\n",
      "NGS: SAMtools \t; IdxStats\n",
      "NGS: SAMtools \t; rmdup\n",
      "NGS: SAMtools \t; MPileup\n",
      "NGS: SAMtools \t; Stats\n",
      "NGS: SAMtools \t; Generate pileup\n",
      "NGS: SAMtools \t; Generate pileup\n",
      "NGS: SAMtools \t; Filter SAM\n",
      "NGS: VCF Manipulation \t; VCFflatten:\n",
      "NGS: VCF Manipulation \t; VCFfilter:\n",
      "NGS: VCF Manipulation \t; VCFannotateGenotypes:\n",
      "NGS: VCF Manipulation \t; VCFgenotypes:\n",
      "NGS: VCF Manipulation \t; VCFprimers:\n",
      "NGS: VCF Manipulation \t; VCFannotate:\n",
      "NGS: VCF Manipulation \t; VCF-VCFintersect:\n",
      "NGS: VCF Manipulation \t; VCFprimers:\n",
      "NGS: VCF Manipulation \t; VCFdistance:\n",
      "NGS: VCF Manipulation \t; VCFcheck:\n",
      "NGS: VCF Manipulation \t; VCFaddinfo:\n",
      "NGS: VCF Manipulation \t; VCFgenotype-to-haplotype:\n",
      "NGS: VCF Manipulation \t; VCFcombine:\n",
      "NGS: VCF Manipulation \t; VCFannotateGenotypes:\n",
      "NGS: VCF Manipulation \t; VCFselectsamples:\n",
      "NGS: VCF Manipulation \t; VCFsort:\n",
      "NGS: VCF Manipulation \t; VCFcheck:\n",
      "NGS: VCF Manipulation \t; VCFcommonSamples:\n",
      "NGS: VCF Manipulation \t; VCFfixup:\n",
      "NGS: VCF Manipulation \t; VCFrandomSample:\n",
      "NGS: VCF Manipulation \t; VCFleftAlign:\n",
      "NGS: VCF Manipulation \t; VCF-BEDintersect:\n",
      "NGS: VCF Manipulation \t; VCFselectsamples:\n",
      "NGS: VCF Manipulation \t; VCFhetHomAlleles:\n",
      "NGS: VCF Manipulation \t; VCF-BEDintersect:\n",
      "NGS: VCF Manipulation \t; VCFtoTab-delimited:\n",
      "NGS: VCF Manipulation \t; VCFsort:\n",
      "NGS: VCF Manipulation \t; VCFbreakCreateMulti:\n",
      "NGS: VCF Manipulation \t; VCFfixup:\n",
      "NGS: VCF Manipulation \t; VCFtoTab-delimited:\n",
      "NGS: VCF Manipulation \t; VCFannotate:\n",
      "NGS: VCF Manipulation \t; VCFhetHomAlleles:\n",
      "NGS: VCF Manipulation \t; VcfAllelicPrimitives:\n",
      "NGS: VCF Manipulation \t; VCFgenotypes:\n",
      "NGS: VCF Manipulation \t; VcfAllelicPrimitives:\n",
      "NGS: VCF Manipulation \t; VCFgenotype-to-haplotype:\n",
      "NGS: VCF Manipulation \t; VCFbreakCreateMulti:\n",
      "NGS: VCF Manipulation \t; VCFfilter:\n",
      "NGS: VCF Manipulation \t; VCF-VCFintersect:\n",
      "NGS: VCF Manipulation \t; VCFleftAlign:\n",
      "NGS: VCF Manipulation \t; VCFrandomSample:\n",
      "NGS: VCF Manipulation \t; VCFcommonSamples:\n",
      "NGS: VCF Manipulation \t; VCFcombine:\n",
      "NGS: VCF Manipulation \t; VCFaddinfo:\n",
      "NGS: VCF Manipulation \t; VCFdistance:\n",
      "NGS: Variant Analysis \t; ANNOVAR Annotate VCF\n",
      "NGS: Variant Analysis \t; Varscan\n",
      "NGS: Variant Analysis \t; CloudMap: Hawaiian Variant Mapping with WGS data\n",
      "NGS: Variant Analysis \t; Annotate\n",
      "NGS: Variant Analysis \t; CloudMap: in silico complementation\n",
      "NGS: Variant Analysis \t; Naive Variant Caller\n",
      "NGS: Variant Analysis \t; BamLeftAlign\n",
      "NGS: Variant Analysis \t; FASTA from allele counts\n",
      "NGS: Variant Analysis \t; Variant Annotator\n",
      "NGS: Variant Analysis \t; Slice VCF\n",
      "NGS: Variant Analysis \t; FreeBayes\n",
      "NGS: Variant Analysis \t; MAF boxplot\n",
      "NGS: Variant Analysis \t; CloudMap: Variant Discovery Mapping with WGS data\n",
      "NGS: Variant Analysis \t; Phylorelatives\n",
      "NGS: Variant Analysis \t; CloudMap: EMS Variant Density Mapping\n",
      "Operate on Genomic Intervals \t; Subtract\n",
      "Operate on Genomic Intervals \t; Cluster\n",
      "Operate on Genomic Intervals \t; Complement\n",
      "Operate on Genomic Intervals \t; Intersect\n",
      "Operate on Genomic Intervals \t; Join\n",
      "Operate on Genomic Intervals \t; Concatenate\n",
      "Operate on Genomic Intervals \t; Base Coverage\n",
      "Operate on Genomic Intervals \t; Merge\n",
      "Operate on Genomic Intervals \t; Fetch closest non-overlapping feature\n",
      "Operate on Genomic Intervals \t; Get flanks\n",
      "Operate on Genomic Intervals \t; Profile Annotations\n",
      "Operate on Genomic Intervals \t; Coverage\n",
      "Phenotype Association \t; aaChanges\n",
      "Phenotype Association \t; HVIS\n",
      "Phenotype Association \t; SIFT\n",
      "Phenotype Association \t; LD\n",
      "Phenotype Association \t; PASS\n",
      "Phenotype Association \t; DAVID\n",
      "Phenotype Association \t; BEAM\n",
      "Phenotype Association \t; LPS\n",
      "Phenotype Association \t; snpFreq\n",
      "Phenotype Association \t; CTD\n",
      "Phenotype Association \t; g:Profiler\n",
      "Phenotype Association \t; GPASS\n",
      "Phenotype Association \t; MasterVar to pgSnp\n",
      "Phenotype Association \t; FunDO\n",
      "Phenotype Association \t; Call SNPS with Freebayes\n",
      "Phenotype Association \t; Separate pgSnp alleles\n",
      "Phenotype Association \t; Annotate with DGI\n",
      "Phenotype Association \t; phyloP\n",
      "Phenotype Association \t; VCF to pgSnp\n",
      "RNA Structure Prediction \t; Iterative Mapping\n",
      "RNA Structure Prediction \t; RNA Structure Prediction\n",
      "RNA Structure Prediction \t; Reactivity Calculation\n",
      "RNA Structure Prediction \t; Get RT Stop Counts\n",
      "Regional Variation \t; Principal Component Analysis\n",
      "Regional Variation \t; Make windows\n",
      "Regional Variation \t; Kernel Principal Component Analysis\n",
      "Regional Variation \t; Fetch substitutions \n",
      "Regional Variation \t; Mask CpG/non-CpG sites\n",
      "Regional Variation \t; Feature coverage\n",
      "Regional Variation \t; Compute RCVE\n",
      "Regional Variation \t; Heatmap \n",
      "Regional Variation \t; Estimate Indel Rates\n",
      "Regional Variation \t; Fit HMM \n",
      "Regional Variation \t; Perform Linear Regression\n",
      "Regional Variation \t; Fetch Indels\n",
      "Regional Variation \t; Compute partial R square\n",
      "Regional Variation \t; Estimate substitution rates \n",
      "Regional Variation \t; Filter nucleotides\n",
      "Regional Variation \t; Perform Logistic Regression with vif\n",
      "Regional Variation \t; Perform Best-subsets Regression\n",
      "Regional Variation \t; Canonical Correlation Analysis\n",
      "Regional Variation \t; Extract Orthologous Microsatellites\n",
      "Regional Variation \t; Assign weighted-average\n",
      "Regional Variation \t; Estimate microsatellite mutability\n",
      "Regional Variation \t; Kernel Canonical Correlation Analysis\n",
      "Regional Variation \t; Fetch Indels\n",
      "Regional Variation \t; Karyotype Plotting tool\n",
      "Send Data \t; GenomeSpace Exporter\n",
      "Statistics \t; Draw ROC plot\n",
      "Statistics \t; Summary Statistics\n",
      "Statistics \t; Correlation\n",
      "Statistics \t; Count\n",
      "Statistics \t; Generate A Matrix\n",
      "Statistics \t; Perform LDA\n",
      "Text Manipulation \t; Select last\n",
      "Text Manipulation \t; Paste\n",
      "Text Manipulation \t; Trim\n",
      "Text Manipulation \t; Create single interval\n",
      "Text Manipulation \t; Secure Hash / Message Digest\n",
      "Text Manipulation \t; Convert\n",
      "Text Manipulation \t; Line/Word/Character count\n",
      "Text Manipulation \t; Cut\n",
      "Text Manipulation \t; Select first\n",
      "Text Manipulation \t; Add column\n",
      "Text Manipulation \t; Select random lines\n",
      "Text Manipulation \t; Change Case\n",
      "Text Manipulation \t; Condense\n",
      "Text Manipulation \t; Merge Columns\n",
      "Text Manipulation \t; Concatenate datasets\n",
      "Text Manipulation \t; Remove beginning\n",
      "Text Manipulation \t; Compute\n",
      "snpEff \t; SnpSift Filter\n",
      "snpEff \t; SnpEff\n",
      "snpEff \t; CloudMap: Check snpEff Candidates\n"
     ]
    }
   ],
   "source": [
    "import bioblend.galaxy\n",
    "\n",
    "def get_tools(server, api_key):\n",
    "    gi = bioblend.galaxy.GalaxyInstance(url=server, key=api_key)\n",
    "    gi.tools.get_tools()\n",
    "    tools = gi.tools.get_tools()\n",
    "    sections=set()\n",
    "    for element in tools:\n",
    "        sections.add(element['panel_section_name'])\n",
    "    sorted_sections = list(sections)\n",
    "    sorted_sections.sort()\n",
    "    #print sorted_sections\n",
    "    for section in sorted_sections:\n",
    "        for element in tools: \n",
    "            if element['panel_section_name']==section:\n",
    "                print section,'\\t;', element['name']\n",
    "                \n",
    "get_tools('https://usegalaxy.org/', '29197b9c690d49fbf511447d4a676ebc')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None \t; Wiggle to Interval\n",
      "None \t; Convert lped to fped\n",
      "None \t; Convert MAF to Genomic Intervals\n",
      "None \t; Convert plink pbed to linkage lped\n",
      "None \t; Convert plink pbed to ld reduced format\n",
      "None \t; Convert Genomic Intervals To Strict BED12\n",
      "None \t; Convert lped to plink pbed\n",
      "None \t; Convert Len file to Linecount\n",
      "None \t; Convert tabular to dbnsfp\n",
      "None \t; Convert Genomic Intervals To BED\n",
      "None \t; Convert Genomic Intervals To Strict BED\n",
      "None \t; Convert GFF to BED\n",
      "None \t; Convert GFF to Feature Location Index\n",
      "None \t; Import History\n",
      "None \t; Convert BED to GFF\n",
      "None \t; Convert MAF to Fasta\n",
      "None \t; Convert FASTA to len file\n",
      "None \t; Convert Genomic Intervals To Strict BED6\n",
      "None \t; Set External Metadata\n",
      "None \t; Convert Genomic Intervals To Coverage\n",
      "None \t; Convert BED to Feature Location Index\n",
      "None \t; Convert FASTA to Bowtie base space Index\n",
      "None \t; Convert FASTA to Tabular\n",
      "None \t; Export History\n",
      "None \t; Convert FASTA to Bowtie color space Index\n",
      "None \t; Convert FASTA to 2bit\n",
      "None \t; Convert SAM to BAM\n",
      "None \t; Convert Picard Interval List to BED6\n",
      "Convert Formats \t; MAF to BED\n",
      "Convert Formats \t; Wig/BedGraph-to-bigWig\n",
      "Convert Formats \t; MAF to FASTA\n",
      "Convert Formats \t; GTF-to-BEDGraph\n",
      "Convert Formats \t; Wiggle-to-Interval\n",
      "Convert Formats \t; AXT to concatenated FASTA\n",
      "Convert Formats \t; SFF converter\n",
      "Convert Formats \t; GFF-to-BED\n",
      "Convert Formats \t; MAF to Interval\n",
      "Convert Formats \t; AXT to LAV\n",
      "Convert Formats \t; LAV to BED\n",
      "Convert Formats \t; BED-to-bigBed\n",
      "Convert Formats \t; BED-to-GFF\n",
      "Convert Formats \t; Tabular-to-FASTA\n",
      "Convert Formats \t; AXT to FASTA\n",
      "Convert Formats \t; FASTA-to-Tabular\n",
      "DNA Motif Tools \t; PSCAN\n",
      "DNA Motif Tools \t; FASTA to Counts Matrix\n",
      "DNA Motif Tools \t; PATSER - Total Hits\n",
      "Diagnostic Tests \t; Diagnostic accuracy\n",
      "Diagnostic Tests \t; Compare-AUC\n",
      "Diagnostic Tests \t; Agreement\n",
      "Extract Features \t; Gene BED To Exon/Intron/Codon BED\n",
      "FASTA manipulation \t; Filter sequences by length\n",
      "FASTA manipulation \t; RNA/DNA\n",
      "FASTA manipulation \t; Compute sequence length\n",
      "FASTA manipulation \t; Collapse\n",
      "FASTA manipulation \t; FASTA Width\n",
      "FASTA manipulation \t; Concatenate\n",
      "Fetch Alignments \t; Stitch Gene blocks\n",
      "Fetch Alignments \t; Extract MAF blocks\n",
      "Fetch Alignments \t; Extract Pairwise MAF blocks\n",
      "Fetch Alignments \t; Extract MAF by block number\n",
      "Fetch Alignments \t; Join MAF blocks\n",
      "Fetch Alignments \t; Reverse Complement\n",
      "Fetch Alignments \t; Filter MAF blocks\n",
      "Fetch Alignments \t; Filter MAF\n",
      "Fetch Alignments \t; Filter MAF blocks\n",
      "Fetch Alignments \t; MAF Coverage Stats\n",
      "Fetch Alignments \t; Stitch MAF blocks\n",
      "Fetch Alignments \t; Split MAF blocks\n",
      "Fetch Sequences \t; Extract Genomic DNA\n",
      "Filter and Sort \t; Extract features\n",
      "Filter and Sort \t; Filter GFF data by attribute\n",
      "Filter and Sort \t; Filter GFF data by feature count\n",
      "Filter and Sort \t; Sort\n",
      "Filter and Sort \t; Filter\n",
      "Filter and Sort \t; Filter GTF data by attribute values_list\n",
      "Filter and Sort \t; Select\n",
      "Filter and Sort \t; Histogram\n",
      "Get Data \t; UCSC Main\n",
      "Get Data \t; Ratmine\n",
      "Get Data \t; MouseMine\n",
      "Get Data \t; CBI Rice Mart\n",
      "Get Data \t; UCSC Archaea\n",
      "Get Data \t; metabolicMine\n",
      "Get Data \t; Upload File\n",
      "Get Data \t; modENCODE fly\n",
      "Get Data \t; BioMart\n",
      "Get Data \t; Get Microbial Data\n",
      "Get Data \t; modENCODE modMine\n",
      "Get Data \t; Flymine\n",
      "Get Data \t; BioMart\n",
      "Get Data \t; EBI SRA\n",
      "Get Data \t; GenomeSpace import\n",
      "Get Data \t; UCSC Test\n",
      "Get Data \t; WormBase\n",
      "Get Data \t; ZebrafishMine\n",
      "Get Data \t; modENCODE worm\n",
      "Get Data \t; EuPathDB\n",
      "Get Data \t; GrameneMart\n",
      "Get Data \t; HbVar\n",
      "Get Data \t; YeastMine\n",
      "Get Data \t; Wormbase\n",
      "Get Data \t; Flymine test\n",
      "Graph/Display Data \t; Plotting tool\n",
      "Graph/Display Data \t; Scatterplot\n",
      "Graph/Display Data \t; Bar chart\n",
      "Graph/Display Data \t; Build custom track\n",
      "Graph/Display Data \t; VCF to MAF Custom Track\n",
      "Graph/Display Data \t; Boxplot\n",
      "Graph/Display Data \t; GMAJ\n",
      "Graph/Display Data \t; Histogram\n",
      "Heatmap Tools \t; heatmap-Microarray Data\n",
      "Heatmap Tools \t; Heatmap\n",
      "Heatmap Tools \t; heatmap-public\n",
      "Join, Subtract and Group \t; Join two Datasets\n",
      "Join, Subtract and Group \t; Group\n",
      "Join, Subtract and Group \t; Subtract Whole Dataset\n",
      "Join, Subtract and Group \t; Compare two Datasets\n",
      "Lift-Over \t; Convert genome coordinates\n",
      "Multiple regression \t; Perform Best-subsets Regression\n",
      "Multivariate Analysis \t; Principal Component Analysis\n",
      "Multivariate Analysis \t; Kernel Canonical Correlation Analysis\n",
      "Multivariate Analysis \t; Canonical Correlation Analysis\n",
      "Multivariate Analysis \t; Kernel Principal Component Analysis\n",
      "NGS: ChIP-seq \t; MACS2 callpeak\n",
      "NGS: ChIP-seq \t; MACS2 bdgdiff\n",
      "NGS: ChIP-seq \t; MACS2 filterdup\n",
      "NGS: ChIP-seq \t; MACS2 bdgbroadcall\n",
      "NGS: ChIP-seq \t; MACS2 bdgpeakcall\n",
      "NGS: ChIP-seq \t; MACS2 randsample\n",
      "NGS: ChIP-seq \t; MACS2 bdgcmp\n",
      "NGS: ChIP-seq \t; MACS2 refinepeak\n",
      "NGS: ChIP-seq \t; MACS2 predictd\n",
      "NGS: SAM Tools \t; Merge BAM Files\n",
      "Operate on Genomic Intervals \t; Get flanks\n",
      "Operate on Genomic Intervals \t; Subtract\n",
      "Operate on Genomic Intervals \t; Cluster\n",
      "Operate on Genomic Intervals \t; Base Coverage\n",
      "Operate on Genomic Intervals \t; Profile Annotations\n",
      "Operate on Genomic Intervals \t; Join\n",
      "Operate on Genomic Intervals \t; Coverage\n",
      "Operate on Genomic Intervals \t; Merge\n",
      "Operate on Genomic Intervals \t; Concatenate\n",
      "Operate on Genomic Intervals \t; Complement\n",
      "Operate on Genomic Intervals \t; Intersect\n",
      "Operate on Genomic Intervals \t; Fetch closest non-overlapping feature\n",
      "Other Tools \t; Convert-Gene-Alias\n",
      "Other Tools \t; arrayLayout\n",
      "Other Tools \t; geneListOverlap\n",
      "Other Tools \t; ddCt\n",
      "Plotting Tools \t; Boxplot/Dotplot\n",
      "Send Data \t; GenomeSpace Exporter\n",
      "Statistics \t; Count\n",
      "Statistics \t; Summary Statistics\n",
      "Survival Analysis \t; Data-Loi2007\n",
      "Survival Analysis \t; RP-Taylor2010\n",
      "Survival Analysis \t; CoxPH-Shedden2008\n",
      "Survival Analysis \t; RP-Shedden2008\n",
      "Survival Analysis \t; RP-Desmedt2007\n",
      "Survival Analysis \t; CoxPH-Loi2007\n",
      "Survival Analysis \t; RP-Glinsky2004\n",
      "Survival Analysis \t; Prognosis-Loi2007\n",
      "Survival Analysis \t; RP-Loi2007\n",
      "Testing Tools \t; Weeder2\n",
      "Text Manipulation \t; Line/Word/Character count\n",
      "Text Manipulation \t; Select random lines\n",
      "Text Manipulation \t; Change Case\n",
      "Text Manipulation \t; Trim\n",
      "Text Manipulation \t; Create single interval\n",
      "Text Manipulation \t; Select last\n",
      "Text Manipulation \t; Condense\n",
      "Text Manipulation \t; Secure Hash / Message Digest\n",
      "Text Manipulation \t; Merge Columns\n",
      "Text Manipulation \t; Convert\n",
      "Text Manipulation \t; Cut\n",
      "Text Manipulation \t; Paste\n",
      "Text Manipulation \t; Concatenate datasets\n",
      "Text Manipulation \t; Select first\n",
      "Text Manipulation \t; Remove beginning\n",
      "Text Manipulation \t; Compute\n",
      "Text Manipulation \t; Add column\n",
      "Unix Tools \t; Uniq\n",
      "Unix Tools \t; Sort\n"
     ]
    }
   ],
   "source": [
    "get_tools('http://galaxy.cruk.cam.ac.uk/','56a7381febed6f914132bdad20854f84 ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
